src/metabase/sync/interface.clj

```diff
@@ -21,15 +21,17 @@
 
 (def TableMetadataField
   "Schema for a given Field as provided in `describe-table`."
-  {:name                           su/NonBlankString
-   :database-type                  (s/maybe su/NonBlankString) ; blank if the Field is all NULL & untyped, i.e. in Mongo
-   :base-type                      su/FieldType
-   :database-position              su/IntGreaterThanOrEqualToZero
-   (s/optional-key :semantic-type) (s/maybe su/FieldType)
-   (s/optional-key :field-comment) (s/maybe su/NonBlankString)
-   (s/optional-key :pk?)           s/Bool
-   (s/optional-key :nested-fields) #{(s/recursive #'TableMetadataField)}
-   (s/optional-key :custom)        {s/Any s/Any}})
+  {:name                               su/NonBlankString
+   :database-type                      (s/maybe su/NonBlankString) ; blank if the Field is all NULL & untyped, i.e. in Mongo
+   :base-type                          su/FieldType
+   :database-position                  su/IntGreaterThanOrEqualToZero
+   (s/optional-key :semantic-type)     (s/maybe su/FieldType)
+   (s/optional-key :effective-type)    (s/maybe su/FieldType)
+   (s/optional-key :coercion-strategy) (s/maybe su/CoercionStrategy)
+   (s/optional-key :field-comment)     (s/maybe su/NonBlankString)
+   (s/optional-key :pk?)               s/Bool
+   (s/optional-key :nested-fields)     #{(s/recursive #'TableMetadataField)}
+   (s/optional-key :custom)            {s/Any s/Any}})
```

Comment:
Just a thought: we should probably make this schema open with `s/Keyword s/Any` so new keys we add in the future don't break things. e.g. if a driver starts returning `:effective-type` it wouldn't work on Metabase < 0.39.0. Not that changing the schema in 0.39.0 would make any difference, but in the future it would be easier for driver authors to take advantage of new features and still support older MB versions